### DISCUSSION TITLE
Reactive Transport with ODML in Reaktoro V2

### DISCUSSION THREAD

**Discussion by mcrossover97** [reaktoro/reaktoro#383] [Q&A]
### Description

Hi Allan,

I hope you are doing well. I am reaching out to inquire if you have any ODML reactive transport examples compatible with Reaktoro V2. I have been unable to find any; the most recent examples are on the v1-odml branch and require modifications to work with Reaktoro V2. Do you have any examples that work with V2, or will I need to modify the old examples to make them compatible?

Best regards,
Mohammad

### What is your operating system?

None

**Comment by allanleal** [comment]
Hi Mohammad, thank you for your question. ODML solvers are best exemplified in reactive transport examples. Reaktoro v1 had a 1D reactive transport solver. This has not yet been ported to Reaktoro v2. The API of ODML-enabled solvers (e.g. `SmartEquilibriumSolver`) is, however, very similar to that of conventional equivalents (e.g. `EquilibriumSolver`). You can find more about their API on https://reaktoro.org/api

**Reply by mcrossover97** [reply]
Thanks, Allan.

It seems the reactive transport examples in the `v1-odml` branch are not functioning properly. When I run the `demo-rt-odml-dolomitization.py` script, I encounter the following error:
```
ODML algorithm :
Traceback (most recent call last):
  File "demo-rt-odml-dolomitization.py", line 991, in <module>
    simulate(params)
  File "demo-rt-odml-dolomitization.py", line 290, in simulate
    smart_equilibrium_options = SmartEquilibriumOptions()
TypeError: reaktoro.PyReaktoro.SmartEquilibriumOptions: No constructor defined!
```

Similarly, I get this error when running the `demo-rtsolver-odml-dolomitization.py` script:
```
Traceback (most recent call last):
  File "demo-rtsolver-odml-dolomitization.py", line 112, in <module>
    smart_method = SmartEquilibriumStrategy.Clustering
NameError: name 'SmartEquilibriumStrategy' is not defined
```

Could you please confirm if there are currently any working reactive transport w/ ODML scripts in any version of Reaktoro? If I'm right, the only working example of ODML is `ex-odml-equilibrium-co2-solubility-brine.py` in Reaktoro V2.

Thanks
